src/TSHelper.ts

```diff
@@ -157,9 +164,16 @@ export function isExplicitArrayType(type: ts.Type, checker: ts.TypeChecker, prog
         return true;
     }
 
+    if (type.symbol) {
+        const baseConstraint = checker.getBaseConstraintOfType(type);
+        if (baseConstraint) {
+            return isExplicitArrayType(baseConstraint, checker, program);
+        }
+    }
+
     const flags = ts.NodeBuilderFlags.InTypeAlias | ts.NodeBuilderFlags.AllowEmptyTuple;
     const typeNode = checker.typeToTypeNode(type, undefined, flags);
-    return typeNode !== undefined && (ts.isArrayTypeNode(typeNode) || ts.isTupleTypeNode(typeNode));
+    return typeNode !== undefined ? ts.isArrayTypeNode(typeNode) || ts.isTupleTypeNode(typeNode) : false;
```

Comment:
I prefer the previous version of this line, why the change?

Reply:
Shouldn't have touched that line, going with preference